proxy/handlers.go

```diff
@@ -32,27 +31,34 @@ func (p *Proxy) registerDashboardHandlers(r *mux.Router) *mux.Router {
 		csrf.CookieName(fmt.Sprintf("%s_csrf", p.cookieOptions.Name)),
 		csrf.ErrorHandler(http.HandlerFunc(httputil.CSRFFailureHandler)),
 	))
+	// dashboard endpoints can be used by user's to view, or modify their session
 	h.HandleFunc("/", p.UserDashboard).Methods(http.MethodGet)
 	h.HandleFunc("/impersonate", p.Impersonate).Methods(http.MethodPost)
 	h.HandleFunc("/sign_out", p.SignOut).Methods(http.MethodGet, http.MethodPost)
 
 	// Authenticate service callback handlers and middleware
+	// callback used to set route-scoped session and redirect back to destination
+	// only accept signed requests (hmac) from other trusted pomerium services
 	c := r.PathPrefix(dashboardURL + "/callback").Subrouter()
-	// only accept payloads that have come from a trusted service (hmac)
 	c.Use(middleware.ValidateSignature(p.SharedKey))
-	c.HandleFunc("/", p.Callback).Queries("redirect_uri", "{redirect_uri}").Methods(http.MethodGet)
 
+	c.Path("/").HandlerFunc(p.ProgrammaticCallback).Methods(http.MethodGet).
```

Comment:
In a future PR, I'd like to split out programmatic login and callback into their own set of handlers and router similar to forward auth.

Reply:
#400